src/main/java/UseCaseClasses/InformationSaver.java

```diff
@@ -65,5 +66,36 @@ public void saveUser(Person user) throws ExecutionException, InterruptedExceptio
         }
         return people;
     }
+
+    public void saveReview(Review review) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        Map<String, Object> data = new HashMap<>();
+        data.put("Location", review.getLocation());
+        data.put("reviewBody", review.getReviewBody());
+        data.put("rating", review.getRating());
+        data.put("FullReview", review.getFullReview());
+        DocumentReference docRef = db.collection("reviews").document(review.getLocation());
+        ApiFuture<WriteResult> result = docRef.set(data);
+        ApiFuture<QuerySnapshot> query = db.collection("reviews").get();
+        QuerySnapshot querySnapshot = query.get(); //
+    }
+
+    public ArrayList<Review> retrieveReviews() throws ExecutionException, InterruptedException, IllegalArgumentException {
+
+        ArrayList<Review> allReviews = new ArrayList<>();
+
+        ApiFuture<QuerySnapshot> query = db.collection("reviews").get();
+        QuerySnapshot querySnapshot = query.get();
+        List<QueryDocumentSnapshot> documents = querySnapshot.getDocuments();
```

Comment:
If you're having trouble with testing, check the pinned comment on the discord server under "code-discussion", and if that still doesn't work, probably you'll have to ask dennis about it